File: INTRODUCCION-POO/EJ_7/EJ_7.py
class Celular:
    def __init__(self):
        self.aplicaciones_instaladas = 0
        self.espacio_disponible = 1024
        self.bateria = 100
    
    def instalar_aplicacion(self, tamaño):
        if self.aplicaciones_instaladas >= 20:
            print("No hay espacio para más aplicaciones")
            return
        if tamaño > self.espacio_disponible:
            print("No hay suficiente espacio de almacenamiento")
            return
        self.aplicaciones_instaladas += 1
        self.espacio_disponible -= tamaño
        print("Aplicación instalada correctamente")
    
    def usar_aplicacion(self, tamaño, minutos):
        if self.bateria <= 0:
            print("Celular apagado")
            return
        
        if tamaño > 250:
            porcentaje_por_10min = 5
        elif tamaño > 100:
            porcentaje_por_10min = 2
        else:
            porcentaje_por_10min = 1
        
        consumo = (minutos // 10) * porcentaje_por_10min
        self.bateria -= consumo
        
        if self.bateria <= 0:
            self.bateria = 0
            print("Celular apagado")
        else:
            print(f"Aplicación usada. Batería restante: {self.bateria}%")
    
    def mostrar_bateria(self):
        print(f"Batería restante: {self.bateria}%")

mi_celular = Celular()

mi_celular.instalar_aplicacion(200)
mi_celular.instalar_aplicacion(50)
mi_celular.instalar_aplicacion(300)

mi_celular.usar_aplicacion(200, 30)
mi_celular.mostrar_bateria()

mi_celular.usar_aplicacion(300, 50)
mi_celular.mostrar_bateria()

mi_celular.usar_aplicacion(50, 100)

File: INTRODUCCION-POO/EJ_7/EJ_7.java
class Celular {
    private int aplicacionesInstaladas;
    private int espacioDisponible;
    private int bateria;
    
    public Celular() {
        this.aplicacionesInstaladas = 0;
        this.espacioDisponible = 1024;
        this.bateria = 100;
    }
    
    public void instalarAplicacion(int tamaño) {
        if (aplicacionesInstaladas >= 20) {
            System.out.println("No hay espacio para más aplicaciones");
            return;
        }
        if (tamaño > espacioDisponible) {
            System.out.println("No hay suficiente espacio de almacenamiento");
            return;
        }
        aplicacionesInstaladas++;
        espacioDisponible -= tamaño;
        System.out.println("Aplicación instalada correctamente");
    }
    
    public void usarAplicacion(int tamaño, int minutos) {
        if (bateria <= 0) {
            System.out.println("Celular apagado");
            return;
        }
        
        int porcentajePor10Min;
        if (tamaño > 250) {
            porcentajePor10Min = 5;
        } else if (tamaño > 100) {
            porcentajePor10Min = 2;
        } else {
            porcentajePor10Min = 1;
        }
        
        int consumo = (minutos / 10) * porcentajePor10Min;
        bateria -= consumo;
        
        if (bateria <= 0) {
            bateria = 0;
            System.out.println("Celular apagado");
        } else {
            System.out.println("Aplicación usada. Batería restante: " + bateria + "%");
        }
    }
    
    public void mostrarBateria() {
        System.out.println("Batería restante: " + bateria + "%");
    }
}

public class MainCelular {
    public static void main(String[] args) {
        Celular miCelular = new Celular();
        
        miCelular.instalarAplicacion(200);
        miCelular.instalarAplicacion(50);
        miCelular.instalarAplicacion(300);
        
        miCelular.usarAplicacion(200, 30);
        miCelular.mostrarBateria();
        
        miCelular.usarAplicacion(300, 50);
        miCelular.mostrarBateria();
        
        miCelular.usarAplicacion(50, 100);
    }
}